### ISSUE TITLE
Deep check seems to ignore state

### ISSUE THREAD

**Issue by bennycornelissen** [wata727/tflint#164]
When running `tflint --deep` on a piece of Terraform code that has already applied successfully, I get errors that suggest tflint ignores state.

```
± tflint --deep
../../../terraform-modules/env-cluster/terraform/master.tf
	ERROR:101 "lab-1-master" is duplicate name. It must be unique. (aws_elb_duplicate_name)
	ERROR:169 "lab-1-scheduler" is duplicate name. It must be unique. (aws_elb_duplicate_name)
../../../terraform-modules/env-cluster/terraform/workers.tf
	ERROR:94 "lab-1-proxy-public" is duplicate name. It must be unique. (aws_elb_duplicate_name)

Result: 3 issues  (3 errors , 0 warnings , 0 notices)
```
These names are in fact duplicate, since the plan has already been applied. If run `terraform destroy` and then try `tflint --deep` again, there are no issues. 

Is this expected/intended behaviour of tflint?

**Comment by wata727** [comment]
Thanks for the report :)

TFLint parses tfstate to check if the resource has already been applied when it checks for a duplicate name of resources. Therefore, this behavior isn't intended. Perhaps it seems that TFLint has failed to detect the tfstate file.

Where is the `terraform.tfstate` in your workspace? Perhaps you can check the log of the tfstate file detection by appending `--debug` at runtime.

**Comment by bennycornelissen** [comment]
I am using remote state (on S3) so there is no local `terraform.tfstate` file.

**Comment by bennycornelissen** [comment]
I just ran tfline with `--debug`: 

```
DEBUG:/loader.go:120 [INFO] Load environment...
DEBUG:/loader.go:128 [ERROR] open .terraform/environment: no such file or directory
DEBUG:/loader.go:131 [INFO] Load tfstate...
DEBUG:/loader.go:135 [ERROR] stat terraform.tfstate: no such file or directory
DEBUG:/loader.go:141 [INFO] Remote state detected
DEBUG:/loader.go:161 [INFO] Load tfvars...
DEBUG:/loader.go:164 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:166 [ERROR] stat terraform.tfvars: no such file or directory
DEBUG:/loader.go:191 [INFO] Load HCL file: `aws-provider.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `aws-provider.tf`
DEBUG:/loader.go:191 [INFO] Load HCL file: `main.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `main.tf`
DEBUG:/loader.go:191 [INFO] Load HCL file: `outputs.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `outputs.tf`
DEBUG:/loader.go:191 [INFO] Load HCL file: `override.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `override.tf`
DEBUG:/loader.go:191 [INFO] Load HCL file: `terraform-backends.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `terraform-backends.tf`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_previous_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_previous_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_not_specified_iam_profile`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_not_specified_iam_profile`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_default_standard_volume`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_default_standard_volume`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_iam_profile`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_iam_profile`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_ami`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_ami`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_key_name`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_key_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_subnet`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_subnet`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_vpc_security_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_invalid_vpc_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_security_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_subnet`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_subnet`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_duplicate_name`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_security_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:153 [INFO] detect by `aws_elb_invalid_security_group`
DEBUG:/aws_elb_invalid_security_group.go:63 [ERROR] ERROR: `"${aws_security_group.cluster-instances.id}"` is not evaluable
DEBUG:/aws_elb_invalid_security_group.go:63 [ERROR] ERROR: `"${aws_security_group.scheduler-elb.id}"` is not evaluable
DEBUG:/aws_elb_invalid_security_group.go:63 [ERROR] ERROR: `"${aws_security_group.elb-public.id}"` is not evaluable
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_subnet`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:153 [INFO] detect by `aws_elb_invalid_subnet`
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${aws_subnet.private.id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_a_subnet_id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_b_subnet_id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_c_subnet_id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_a_subnet_id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_b_subnet_id}"` is not evaluable
DEBUG:/aws_elb_invalid_subnet.go:63 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.public_c_subnet_id}"` is not evaluable
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_instance`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:153 [INFO] detect by `aws_elb_invalid_instance`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_duplicate_name`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:153 [INFO] detect by `aws_elb_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_default_parameter_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_default_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_vpc_security_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_vpc_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_db_subnet_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_db_subnet_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_parameter_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_option_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_option_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_previous_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_previous_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_readable_password`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_readable_password`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_duplicate_identifier`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_duplicate_identifier`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_default_parameter_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_default_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_parameter_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_subnet_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_subnet_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_security_group`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_previous_type`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_previous_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_duplicate_id`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_duplicate_id`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_security_group_duplicate_name`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:153 [INFO] detect by `aws_security_group_duplicate_name`
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/aws_security_group_duplicate_name.go:75 [ERROR] ERROR: `"${data.terraform_remote_state.env-base.vpc_id}"` is not evaluable
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_route_table`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_route_table`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_not_specified_target`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_not_specified_target`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_specified_multiple_targets`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_specified_multiple_targets`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_gateway`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_egress_only_gateway`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_egress_only_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_nat_gateway`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_nat_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_vpc_peering_connection`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_vpc_peering_connection`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_instance`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_instance`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_network_interface`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_network_interface`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_cloudwatch_metric_alarm_invalid_unit`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_cloudwatch_metric_alarm_invalid_unit`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_ecs_cluster_duplicate_name`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_ecs_cluster_duplicate_name`
DEBUG:/detector.go:153 [INFO] detect by `terraform_module_pinned_source`
DEBUG:/detector.go:116 [INFO] detect module `env-cluster`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `terraform_module_pinned_source`
../../../terraform-modules/env-cluster/terraform/master.tf
	ERROR:101 "lab-1-master" is duplicate name. It must be unique. (aws_elb_duplicate_name)
	ERROR:169 "lab-1-scheduler" is duplicate name. It must be unique. (aws_elb_duplicate_name)
../../../terraform-modules/env-cluster/terraform/workers.tf
	ERROR:94 "lab-1-proxy-public" is duplicate name. It must be unique. (aws_elb_duplicate_name)

Result: 3 issues  (3 errors , 0 warnings , 0 notices)
```

**Comment by wata727** [comment]
Thanks for your detailed log.
I considered several possible scenarios, but I could not reproduce this problem...

Terraform also saves the remote state file locally. Please check the `.terraform/terraform.tfstate` file. So please check the following three.

**1. Check the resource identifier of duplicated resource in module**

Please check the identifier of the resource detected by TFLint. In the following example, it will be `aws_elb.bar`.

```hcl
resource "aws_elb" "bar" {
  name = "foobar-terraform-elb" # "foobar-terraform-elb" is duplicate name. It must be unique.

  ...
}
```

**2. Check resource identifiers in tfstate**

By using [jq](https://stedolan.github.io/jq/), you can extract a list of resource identifiers.

```
$ cat .terraform/terraform.tfstate | jq '.modules[].resources | keys'
```

**3. Check resource names in tfstate**

You can extract a list of resource names with the following command:

```
$ cat .terraform/terraform.tfstate | jq '.modules[].resources[].primary.attributes.name'
```

**Comment by endzyme** [comment]
I have similar issues with elasticache redis - detecting "duplicate cluster id" but asset was created and is managed by Terraform (v0.10.4). I also have no `terraform plan` change in flight so not sure why it thinks there's a collision.

**Comment by wata727** [comment]
@apowers Thanks for your advice. It looks like a good proposal.
However, the feature of checking states will be removed in the next release version. Please see [here](https://github.com/wata727/tflint/pull/227).
In the future, your advice will be useful when implementing this feature. Thanks!

**Comment by apowers** [comment]
Sad news; checking for duplicate resources is one of the reasons I want to use tflint. 
Last week when promoting from QA to Prod my terraform plan failed because of a few duplicate resource names and I want to avoid that failure in the future.

I didn't realize that `terraform plan` was intended to precede a deep check.

My workaround to this issue is to `terraform pull` the state into the location that tflint expects to find it before running a deep check. This works well enough for my case.

**Comment by wata727** [comment]
Hi there, in TFLint 0.8.0, rules using tfstate have been temporarily removed. So I will close this issue.
Please try the latest version again. If you still have problems, please open a new issue instead of reopening.
